### DISCUSSION TITLE
About list(DEVICE_DICT.keys()) in other scripts

### DISCUSSION THREAD

**Discussion by BilldarBagdar** [markusschloesser/MackieC4_P3#59] [Q&A]
if `list(DEVICE_DICT.keys())` is just `DEVICE_DICT.keys()`, does it cause you problems in Py3 or whatever?

The return type of keys() is already a List object.  So wrapping with list() seems redundant to me.  But I also don't notice any functional difference one way or the other, so if it does make a difference on your side, it needs to stay.  But if the only reason is because you see it that way in other scripts, I say remove it, it's redundant.

Similarly with the `for_display_only=False` parameter.  It's redundant to use `for_display_only=False` 20 times when we call the method instead of once (as an optional parameter) when we declare the method.  If we ever need to use `for_display_only=True` it will be way easier to spot the "non default behavior" where the method is called with the non-default parameter value `for_display_only=True`  in the one or few non-default places where it is called that way.

**Comment by markusschloesser** [comment]
>  does it cause you problems in Py3 or whatever?

no, it actually doesn't make a difference (currently), but if ALL other scripts (checked 10 or so)  use it like that, I am hesitant, because as my python skills are still lacking, my approach is to see what others are doing and copy that (basically learning like a child).

**Reply by BilldarBagdar** [reply]
Ok - I know it isn't needed, or at least I know enough to be dangerous in that respect.  But I also think I know at most it only matters a few processor cycles either way, and it might even be the kind of thing that gets automatically optimized away by the interpreter/compiler.

[overloading methods](https://docs.python.org/3/faq/programming.html#id60) applies to all methods and specifically mentions "default parameters" and the example is easily translatable to the "optional boolean parameter" case. (from "optional Any parameter")
